### PR TITLE
Delegate pods IPAM to controller manager

### PR DESCRIPTION
**Pull request by fiunchinho**
### What this PR does / why we need it

Towards https://github.com/giantswarm/roadmap/issues/1347

### Checklist

- [X] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/test create

### CODE REVIEW

**Review by whites11 — APPROVED** [review]

File: helm/default-apps-gcp/values.yaml [hunk 1]
```diff
@@ -5,8 +5,17 @@ userConfig:
   cilium:
     configMap:
       values: |
+        ipam:
+          mode: kubernetes
```
Comment (whites11) [inline helm/default-apps-gcp/values.yaml:9]: Be sure controller manager flags are set properly. https://github.com/giantswarm/giantnetes-terraform/blob/master/templates/files/manifests/controller-manager.yaml#L28

Comment (fiunchinho) [reply helm/default-apps-gcp/values.yaml:9]: Doing it here https://github.com/giantswarm/cluster-gcp/pull/141

**Review by AverageMarcus — APPROVED** [review]

**Pull request merged**